src/functions/ladok-events/utils.ts

```
@@ -16,20 +16,10 @@ export function _convert(ugUser: TUgUser): TCourseUser {
   };
 }
 
-export function convertUgToCourseUser(
-  ugUser: TUgUser | undefined,
-): TCourseUser | undefined {
-  if (ugUser === undefined) return undefined;
-  return _convert(ugUser);
-}
-
-export function convertUgToCourseUserArr(
-  ugUser: (TUgUser | undefined)[] | undefined,
+export function convertUgUsersToCourseUsers(
+  ugUser: (TUgUser | undefined)[],
 ): TCourseUser[] {
-  if (Array.isArray(ugUser)) {
-    return (ugUser.filter((o) => o) as TUgUser[]).map(_convert);
-  }
-  return [];
+  return ugUser.filter((o): o is TUgUser => o !== undefined).map(_convert);
```

I think the return type "o is TUgUser" could be confusing since a filter function should evaluate "truthy".

---

That is a good point. `obj is T` is used in TypeScript for functions that return "booleans with type meaning".

```ts
// For example, if the function is declared separately:
function isNotUndefined(obj: number | undefined): obj is number {
  return obj !== undefined;
}

const arr1 = [1, 2, 3, undefined]; // arr1: number | undefined
const arr2 = arr1.filter(isNotUndefined); // We don't need to type-cast!
```

In this particular case, your option is more readable and doesn't make any difference, so I'll put it back!

---

I think we should throw an error in getUgUser because that would make sense: "we expected to get user X from UG but got nothing" and UG will be a good starting point for troubleshooting.